scripts/bootstrap.js

```diff
@@ -109,30 +110,46 @@ async function * downloader (key, all) {
   runtimes = runtimes.checkout(runtimes.version)
   goodbye(() => runtimes.close())
 
-  yield `\n  Extracting platform runtime${all ? 's' : ''} to disk\n`
+  console.log(`\n  Extracting platform runtime${all ? 's' : ''} to disk\n`)
 
   const runtime = runtimes.mirror(new Localdrive(SWAP), {
     prefix: '/by-arch' + (all ? '' : '/' + ADDON_HOST)
   })
 
   for await (const { op, key, bytesAdded } of runtime) {
     if (op === 'add') {
-      yield '\x1B[32m+\x1B[39m ' + key + ' [' + byteSize(bytesAdded) + ']'
+      console.log('\x1B[32m+\x1B[39m ' + key + ' [' + byteSize(bytesAdded) + ']')
     } else if (op === 'change') {
-      yield '\x1B[33m~\x1B[39m ' + key + ' [' + byteSize(bytesAdded) + ']'
+      console.log('\x1B[33m~\x1B[39m ' + key + ' [' + byteSize(bytesAdded) + ']')
     } else if (op === 'remove') {
-      yield '\x1B[31m-\x1B[39m ' + key + ' [' + byteSize(bytesAdded) + ']'
+      console.log('\x1B[31m-\x1B[39m ' + key + ' [' + byteSize(bytesAdded) + ']')
     }
   }
 
-  yield '\x1B[2K\x1B[200D  Runtime extraction complete\x1b[K\n'
+  console.log('\x1B[2K\x1B[200D  Runtime extraction complete\x1b[K\n')
 
   await runtimes.close()
   await swarm.destroy()
   await corestore.close()
 
   const tick = isWindows ? '^' : '✔'
 
-  if (all) yield '\x1B[32m' + tick + '\x1B[39m Download complete\n'
-  else yield '\x1B[32m' + tick + '\x1B[39m Download complete, initalizing...\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n'
+  if (all) console.log('\x1B[32m' + tick + '\x1B[39m Download complete\n')
+  else console.log('\x1B[32m' + tick + '\x1B[39m Download complete, initalizing...\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n')
+}
+
+/**
+ * @param {Hyperdrive} drive
+ */
+async function monitorDrive (drive) {
+  const downloadSpeedometer = speedometer()
+  let downloadedBytes = 0
+  const blobs = await drive.getBlobs()
+  blobs.core.on('download', (_index, bytes) => {
+    downloadedBytes += bytes
+    const speed = downloadSpeedometer(bytes)
+    process.stdout.clearLine();
+    process.stdout.cursorTo(0);
+    process.stdout.write(`Downloaded: ${byteSize(downloadedBytes)} - Speed: ${byteSize(speed)}/s`);
```

Comment:
dont update ui here, simply update the speed function and run a timer every 500ms that updates the ui with `downloadSpeedmeter()`, also track uploads also